clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/view/AmazonSecurityGroupProvider.groovy

```diff
@@ -64,27 +64,42 @@ class AmazonSecurityGroupProvider implements SecurityGroupProvider<AmazonSecurit
   }
 
   @Override
-  Set<AmazonSecurityGroup> getAll(boolean includeRules) {
-    getAllMatchingKeyPattern(Keys.getSecurityGroupKey('*', '*', '*', '*', '*'), includeRules)
+  Collection<AmazonSecurityGroup> getAll(boolean includeRules) {
+    if (!includeRules) {
+      def identifiers = cacheView.getIdentifiers(SECURITY_GROUPS.ns)
```

Comment:
@anotherchrisberry actually did this as part of https://github.com/spinnaker/clouddriver/pull/2533 ... my contribution was testing and realization that `List` -> `Set` actually takes ~6s given our volume of data.